lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -829,6 +830,7 @@ public function where($predicates)
     public function andWhere($where)
     {
         $args  = func_get_args();
+        $args  = array_filter($args); // TODO remove for 3.0 https://github.com/doctrine/dbal/issues/4282
```

Comment:
Instead of a TODO, it's better to file a release blocker for 3.0. It's fine to leave a comment explaining why this is needed though.

Reply:
I don't know what a release blocker is. Is it an issue that mentions that this code should be removed?

Reply:
Yes, please file an issue. I'll mark it as a blocker and assign it to the 3.0.0 milestone. This way, we'll have it tracked in the issue tracker, not in the source code.

Reply:
Anyway I've removed the TODO now. Should I squash?

Reply:
Yep, looks good. I also ran the test on the previous release to make the patch doesn't fix something that's not broken:
```
$ git checkout 2.10.2
HEAD is now at aab745e7b Release v2.10.2

$ git diff 2.11.x 19f33f7072cbbd6edf103551298ff12a068209c2 -- tests | patch
patching file tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php

$ phpunit tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php
PHPUnit 8.4.1 by Sebastian Bergmann and contributors.

.............................................................................     77 / 77 (100%)

Time: 1 second, Memory: 12.00 MB

OK (77 tests, 111 assertions)
```

Reply:
Squashed!